symbols/overlay29.yml

```diff
@@ -771,6 +821,17 @@ overlay29:
         r2: y position
         r3: flags
         stack[0]: visibility flag
+    - name: PerformLeaderAction
+      address:
+        EU: 0x22F1890
+        NA: 0x22F0EDC
+        JP: 0x22F24DC
+      description: |-
+        This appears to be the function that actually performs the leader's action within RunLeaderTurn.
```

Comment:
This is either incorrect or incomplete, this function sets the leader's action field depending on the current situation (for example, depending on which buttons are currently pressed). It returns once the leader has an action assigned, so said action is probably executed later on.

Reply:
Hmm are you sure? I admittedly didn't look too much at this function (it was just call "LeaderTurn" in irdkwia's spreadsheet), but it's quite a complex function. There's a big while loop in it that starts with `GetLeader` and then does a bunch of stuff.

On top of that, the way it's called within `RunLeaderTurn` seems pretty suggestive, although I'm sort of confused how it fits in with `ExecuteMonsterAction`:
```c
// excerpt
...
    bVar2 = IsFloorOver();
    if (bVar2 != '\0') {
      return '\0';
    }
    FUN_022e81f8();
    EnemyEvolution(peVar3);
    (*PTR_DUNGEON_PTR_022ec600)->no_action_in_progress = '\x01';
    LeaderTurn();
    (*PTR_DUNGEON_PTR_022ec600)->no_action_in_progress = '\0';
    bVar2 = IsFloorOver();
    if (bVar2 != '\0') {
      return '\x01';
    }
    ExecuteMonsterAction(peVar3);
    FUN_022ef9c8();
    TryForcedLoss('\0');
    bVar2 = IsFloorOver();
    if (bVar2 != '\0') {
      return '\x01';
    }
    puVar5 = (ushort *)peVar3->info;
    if ((*puVar5 & 0x8000) != 0) {
      *puVar5 = *puVar5 & 0x7fff;
      *puVar5 = *puVar5 | 0x4000;
    }
    bVar2 = IsFloorOver();
    ppdVar1 = PTR_DUNGEON_PTR_022ec600;
    if (bVar2 != '\0') {
      return '\x01';
    }
...
```

If what you say is true, I suppose that means `dungeon::no_action_in_progress` is probably wrong as well?

Reply:
Oh lol I just realized I was mixing up what **no**_action_in_progress means. I guess your explanation does seem to be in line with what the code above is doing.

I was just guessing while migrating. Could you put up a PR to fix the description? I don't actually have a good idea what this function does.

Reply:
Sure, I'll see if I can come up with a better name and description. The function does multiple things (and I haven't even fully researched it).